packages/web/src/hooks/useManagedAccountNotAllowedRedirect.ts

```diff
@@ -48,7 +53,46 @@ export const useManagedAccountNotAllowedCallback = ({
   useEffect(() => {
     if (isManagedAccount && trigger) {
       callback()
-      toast(messages.unauthorized)
+      toast(messages.unauthorizedAsManaged)
     }
   }, [isManagedAccount, callback, toast, trigger])
 }
+
+/**
+ * Hook to prevent a managed account from some action if the
+ * managed account does not have access to the provided account
+ * @param handle handle of the user we are doing something on behalf of
+ * @param route route to redirect the user to
+ */
+export const useManagedAccountForHandleNotAllowedRedirect = (
+  handle: string,
+  route: string = FEED_PAGE
```

Comment:
should we just route them back by default so people don't lose their work? or does that risk too many infinite loops and confusing scenarios? feed definitely safer

Reply:
Yeah it's a good callout. I'm worried that most users would just end up here by accidentally copying and pasting a link and sending it to someone else, and it will redirect them before they're unable to type anything. I think ok for now, but def down to revisit!